core/base/matrix_data.hpp

```diff
@@ -289,6 +289,33 @@ struct matrix_data {
         return res;
     }
 
+    /**
+     * Initializes a checkered matrix.
+     *
+     * @param num_block_rows  number of block-rows
+     * @param num_block_cols  number of block-columns
+     * @param diag_block  matrix used to fill diagonal blocks
```

Comment:
This works for non-diagonal blocks as well right ?

Reply:
only for non-block-diagonal. It takes a block and duplicates it to get a complete matrix full of these blocks.